### ISSUE TITLE
Widgets often not showing up

### ISSUE THREAD

**Issue by Diskutant** [felixhageloh/uebersicht#262]
I think this bug was introduced in the last update. 
Often the widgets just don't show up. In the Übersicht menu the widgets are not listed, refresh doesn't help, restart Übersicht doesn't help, logout and login the user in macOS doesn't help, restart the Mac finally helps.

Yesterday I had the weird case that refresh Widgets actually did something... it tried to load the widgets from another user which wasn't logged in.

I assume it has something to do with the different use cases
MacBook Pro Retina Mid 2012 Sierra 10.12.2

User A: at work, 2 displays, widgets showing on the main screen
User B: at home after work, sometimes only the internal display, sometimes 2 external displays, sometimes 2 external displays plus the internal display. 

I always log out the user when leaving work or home.

**Comment by felixhageloh** [comment]
I haven't used Übersicht in a multi user environment much. I will see if I can reproduce this

**Comment by Diskutant** [comment]
Anything new on this one? Can I do something to help?

**Comment by felixhageloh** [comment]
Yes actually! I haven't been able reproduce this error (I've tested with a different user account), but I do have a few questions: 

- do both users use the same widget folder?
- do you see any errors in the debug console and / or the Console app?

**Comment by Diskutant** [comment]
No, they use their own widget folder (default location) with their own copies of widgets.

The debug console doesn't show up.
Console app 

```
standard	08:33:57.524850 +0200	Übersicht	starting server task
standard	08:33:59.250078 +0200	Übersicht	watching /Users/<username>/Library/Application Support/Übersicht/widgets
standard	08:34:01.129117 +0200	Übersicht	starting server task
standard	08:34:03.021508 +0200	Übersicht	watching /Users/<username>/Library/Application Support/Übersicht/widgets
standard	08:34:14.042463 +0200	Übersicht	starting server task
standard	08:34:15.951893 +0200	Übersicht	watching /Users/<username>/Library/Application Support/Übersicht/widgets
```


it goes on and on.

When I quit Übersicht:

`standard	08:36:23.364571 +0200	kernel	[Übersicht pid 5368 mux-aware] exiting, non-mux-aware app count 0, runtime: 0:09:41.730
`

Then I tried starting Übersicht again:

```
standard	08:36:43.527531 +0200	kernel	[Übersicht pid:5663 mux-aware] starting, non-mux-aware app count 0, EG power-state: 0 3D: busy HDA: busy
standard	08:36:43.599664 +0200	Übersicht	starting server task
standard	08:36:45.491005 +0200	Übersicht	watching /Users/<username>/Library/Application Support/Übersicht/widgets
standard	08:36:46.514924 +0200	Übersicht	starting server task
standard	08:36:48.478340 +0200	Übersicht	watching /Users/<username>/Library/Application Support/Übersicht/widgets
```

and those 2 lines are repeating over and over again.

**Comment by felixhageloh** [comment]
from the log it looks like there is an error starting the server task and the app keeps retrying. I have a new build here with better error logging:

https://www.dropbox.com/s/kcjohvajbdymu6m/Uebersicht-1.2.49.app.zip

can you run it and check the console again? Hopefully the actual error will be visible now

**Comment by Diskutant** [comment]
```
standard	08:41:52.610853 +0200	Übersicht	starting server task
standard	08:41:53.413564 +0200	Übersicht	Error loading webview: Error Domain=NSURLErrorDomain Code=-1004 "Could not connect to the server." UserInfo={NSErrorFailingURLKey=http://127.0.0.1:41418/188923479, _kCFStreamErrorDomainKey=1, NSLocalizedDescription=Could not connect to the server., _WKRecoveryAttempterErrorKey=<WKReloadFrameErrorRecoveryAttempter: 0x618000238260>, NSErrorFailingURLStringKey=http://127.0.0.1:41418/188923479, NSUnderlyingError=0x61800045e2a0 {Error Domain=kCFErrorDomainCFNetwork Code=-1004 "(null)" UserInfo={_kCFStreamErrorCodeKey=61, _kCFStreamErrorDomainKey=1}}, _kCFStreamErrorCodeKey=61}
standard	08:41:53.413954 +0200	Übersicht	Error loading webview: Error Domain=NSURLErrorDomain Code=-1004 "Could not connect to the server." UserInfo={NSErrorFailingURLKey=http://127.0.0.1:41418/458558273, _kCFStreamErrorDomainKey=1, NSLocalizedDescription=Could not connect to the server., _WKRecoveryAttempterErrorKey=<WKReloadFrameErrorRecoveryAttempter: 0x618000238980>, NSErrorFailingURLStringKey=http://127.0.0.1:41418/458558273, NSUnderlyingError=0x61800045e1b0 {Error Domain=kCFErrorDomainCFNetwork Code=-1004 "(null)" UserInfo={_kCFStreamErrorCodeKey=61, _kCFStreamErrorDomainKey=1}}, _kCFStreamErrorCodeKey=61}
standard	08:41:54.522330 +0200	Übersicht	watching /Users/aldekamp/Library/Application Support/Übersicht/widgets
standard	08:41:54.542911 +0200	Übersicht	server started on port 41418
standard	08:41:54.545604 +0200	Übersicht	using 2 screens
standard	08:41:55.558404 +0200	Übersicht	starting server task
standard	08:41:57.466386 +0200	Übersicht	watching /Users/aldekamp/Library/Application Support/Übersicht/widgets
standard	08:41:57.488871 +0200	Übersicht	server started on port 41418
standard	08:41:57.491481 +0200	Übersicht	using 2 screens
standard	08:41:58.422216 +0200	Übersicht	Error loading webview: Error Domain=NSURLErrorDomain Code=-1004 "Could not connect to the server." UserInfo={NSErrorFailingURLKey=http://127.0.0.1:41418/188923479, _kCFStreamErrorDomainKey=1, NSLocalizedDescription=Could not connect to the server., _WKRecoveryAttempterErrorKey=<WKReloadFrameErrorRecoveryAttempter: 0x6000004235c0>, NSErrorFailingURLStringKey=http://127.0.0.1:41418/188923479, NSUnderlyingError=0x6100006448f0 {Error Domain=kCFErrorDomainCFNetwork Code=-1004 "(null)" UserInfo={_kCFStreamErrorCodeKey=61, _kCFStreamErrorDomainKey=1}}, _kCFStreamErrorCodeKey=61}
standard	08:41:58.422688 +0200	Übersicht	Error loading webview: Error Domain=NSURLErrorDomain Code=-1004 "Could not connect to the server." UserInfo={NSErrorFailingURLKey=http://127.0.0.1:41418/458558273, _kCFStreamErrorDomainKey=1, NSLocalizedDescription=Could not connect to the server., _WKRecoveryAttempterErrorKey=<WKReloadFrameErrorRecoveryAttempter: 0x600000421a80>, NSErrorFailingURLStringKey=http://127.0.0.1:41418/458558273, NSUnderlyingError=0x600000650a10 {Error Domain=kCFErrorDomainCFNetwork Code=-1004 "(null)" UserInfo={_kCFStreamErrorCodeKey=61, _kCFStreamErrorDomainKey=1}}, _kCFStreamErrorCodeKey=61}

```

repeating over and over again.

**Comment by Diskutant** [comment]
I restarted the Mac, the user from above ran Übersicht correctly then I logged out and logged into the other user and the Console shows some more messages:
```
standard	18:01:55.708885 +0200	Übersicht	starting server task
standard	18:01:57.165014 +0200	Übersicht	Error loading webview: Error Domain=NSURLErrorDomain Code=-1004 "Could not connect to the server." UserInfo={NSErrorFailingURLKey=http://127.0.0.1:41418/724059285, _kCFStreamErrorDomainKey=1, NSLocalizedDescription=Could not connect to the server., _WKRecoveryAttempterErrorKey=<WKReloadFrameErrorRecoveryAttempter: 0x610000229ce0>, NSErrorFailingURLStringKey=http://127.0.0.1:41418/724059285, NSUnderlyingError=0x6100000590b0 {Error Domain=kCFErrorDomainCFNetwork Code=-1004 "(null)" UserInfo={_kCFStreamErrorCodeKey=61, _kCFStreamErrorDomainKey=1}}, _kCFStreamErrorCodeKey=61}
standard	18:01:57.165447 +0200	Übersicht	Error loading webview: Error Domain=NSURLErrorDomain Code=-1004 "Could not connect to the server." UserInfo={NSErrorFailingURLKey=http://127.0.0.1:41418/724053007, _kCFStreamErrorDomainKey=1, NSLocalizedDescription=Could not connect to the server., _WKRecoveryAttempterErrorKey=<WKReloadFrameErrorRecoveryAttempter: 0x61800022ca20>, NSErrorFailingURLStringKey=http://127.0.0.1:41418/724053007, NSUnderlyingError=0x618000247fe0 {Error Domain=kCFErrorDomainCFNetwork Code=-1004 "(null)" UserInfo={_kCFStreamErrorCodeKey=61, _kCFStreamErrorDomainKey=1}}, _kCFStreamErrorCodeKey=61}
standard	18:01:59.100956 +0200	Übersicht	watching /Users/<username>/Library/Application Support/Übersicht/widgets
standard	18:01:59.131050 +0200	Übersicht	server started on port 41418
standard	18:01:59.131269 +0200	Übersicht	Stream client bypassing proxies on TCP Conn [22:0x6080003997e0]
standard	18:01:59.131303 +0200	Übersicht	TCP Conn 0x6080003997e0 started
standard	18:01:59.132215 +0200	Übersicht	TCP Conn 0x6080003997e0 event 1. err: 0
standard	18:01:59.132242 +0200	Übersicht	TCP Conn 0x6080003997e0 complete. fd: 43, err: 0
standard	18:01:59.133946 +0200	Übersicht	using 2 screens
standard	18:02:00.148605 +0200	Übersicht	starting server task
standard	18:02:02.172289 +0200	Übersicht	Error loading webview: Error Domain=NSURLErrorDomain Code=-1004 "Could not connect to the server." UserInfo={NSErrorFailingURLKey=http://127.0.0.1:41418/724059285, _kCFStreamErrorDomainKey=1, NSLocalizedDescription=Could not connect to the server., _WKRecoveryAttempterErrorKey=<WKReloadFrameErrorRecoveryAttempter: 0x618000234740>, NSErrorFailingURLStringKey=http://127.0.0.1:41418/724059285, NSUnderlyingError=0x6180002483a0 {Error Domain=kCFErrorDomainCFNetwork Code=-1004 "(null)" UserInfo={_kCFStreamErrorCodeKey=61, _kCFStreamErrorDomainKey=1}}, _kCFStreamErrorCodeKey=61}
standard	18:02:02.172741 +0200	Übersicht	Error loading webview: Error Domain=NSURLErrorDomain Code=-1004 "Could not connect to the server." UserInfo={NSErrorFailingURLKey=http://127.0.0.1:41418/724053007, _kCFStreamErrorDomainKey=1, NSLocalizedDescription=Could not connect to the server., _WKRecoveryAttempterErrorKey=<WKReloadFrameErrorRecoveryAttempter: 0x610000225cc0>, NSErrorFailingURLStringKey=http://127.0.0.1:41418/724053007, NSUnderlyingError=0x6100000596e0 {Error Domain=kCFErrorDomainCFNetwork Code=-1004 "(null)" UserInfo={_kCFStreamErrorCodeKey=61, _kCFStreamErrorDomainKey=1}}, _kCFStreamErrorCodeKey=61}
standard	18:02:02.244011 +0200	Übersicht	watching /Users/<username>/Library/Application Support/Übersicht/widgets
standard	18:02:02.265804 +0200	Übersicht	server started on port 41418
standard	18:02:02.266046 +0200	Übersicht	Stream client bypassing proxies on TCP Conn [23:0x608000399980]
standard	18:02:02.266071 +0200	Übersicht	TCP Conn 0x608000399980 started
standard	18:02:02.267004 +0200	Übersicht	TCP Conn 0x608000399980 event 1. err: 0
standard	18:02:02.267030 +0200	Übersicht	TCP Conn 0x608000399980 complete. fd: 44, err: 0
standard	18:02:02.268680 +0200	Übersicht	using 2 screens

```

**Comment by felixhageloh** [comment]
Thanks for the console logs! I think it pointed me to some issues how the app handles errors - I will need to look into it some more.

In the meantime, I am wondering if you have to instances of Übersicht running at the same time. You could run 

```
ps ax | grep sicht
```

in the terminal to check

**Comment by felixhageloh** [comment]
I think I found and solved the issue. I need to do some more testing but here is a temporary build that should fix the issue for you already: 

https://www.dropbox.com/s/jve2rnp8n1mlh5e/Uebersicht-1.2.50.app.zip

**Comment by Diskutant** [comment]
Thanks a lot! That version works now.

I checked ps ax | grep sicht after replacing the old version with the new version and then restarting the mac, logged into user a, later logged out, and logged in with user b. 
The result:

```
  755   ??  S      0:17.52 /Applications/Übersicht.app/Contents/Resources/localnode /Applications/Übersicht.app/Contents/Resources/server.js -d /Users/<user a>/Library/Application Support/Übersicht/widgets -p 41416 -s /Users/<user a>/Library/Application Support/tracesOf.Uebersicht
22595   ??  S      0:01.09 /Applications/Übersicht.app/Contents/MacOS/Übersicht
22707   ??  S      0:04.78 /Applications/Übersicht.app/Contents/Resources/localnode /Applications/Übersicht.app/Contents/Resources/server.js -d /Users/<user b>/Library/Application Support/Übersicht/widgets -p 41418 -s /Users/<user b>/Library/Application Support/tracesOf.Uebersicht
26039 s007  S+     0:00.00 grep --color=auto sich
```

looks like on log out Übersicht doesn't quit it's server correctly?

**Comment by felixhageloh** [comment]
Yeah indeed, that was part of the problem. Working on a version that will fix this as well!

**Comment by felixhageloh** [comment]
This should be fixed in 1.2.51 released today
